chia/farmer/farmer.py

```diff
@@ -327,16 +334,19 @@ async def plot_sync_callback(self, peer_id: bytes32, delta: Optional[Delta]) ->
         if receiver.initial_sync() or harvester_updated:
             self.state_changed("harvester_update", receiver.to_dict(True))
 
-    async def _pool_get_pool_info(self, pool_config: PoolWalletConfig) -> Optional[Dict[str, Any]]:
+    async def _pool_get_pool_info(self, pool_config: PoolWalletConfig) -> Optional[GetPoolInfoResult]:
         try:
             async with aiohttp.ClientSession(trust_env=True) as session:
-                async with session.get(
-                    f"{pool_config.pool_url}/pool_info", ssl=ssl_context_for_root(get_mozilla_ca_crt(), log=self.log)
-                ) as resp:
+                url = f"{pool_config.pool_url}/pool_info"
+                async with session.get(url, ssl=ssl_context_for_root(get_mozilla_ca_crt(), log=self.log)) as resp:
                     if resp.ok:
                         response: Dict[str, Any] = json.loads(await resp.text())
                         self.log.info(f"GET /pool_info response: {response}")
-                        return response
+                        new_pool_url: Optional[str] = None
+                        if resp.url != url:
```

Comment:
HTTP redirects come in different flavours. Only in the cases of `301 Moved Permanently` and `308 Permanent Redirect` does it seem reasonable to persist the new URL, and use it for future queries/posts.
e.g. `302 Found` (described as "moved temporarily") would not make sense to persist.

Reply:
Sure, i can restrict by status code